### PR TITLE
Add app

### PR DESCRIPTION
**Pull request by al1img**

### CODE REVIEW

File: src/app/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#
+# Copyright (C) 2024 Renesas Electronics Corporation.
```
Comment (mlohvynenko) [inline src/app/CMakeLists.txt:2]:
```suggestion
```

File: src/app/aoscore.cpp [hunk 2]
```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
```
Comment (mlohvynenko) [inline src/app/aoscore.cpp:2]:
```suggestion
```

File: src/app/aoscore.hpp [hunk 3]
```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
```
Comment (mlohvynenko) [inline src/app/aoscore.hpp:2]:
```suggestion
```

File: src/app/app.cpp [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+}
+
+void App::HandleVersion(const std::string& name, const std::string& value)
+{
+    (void)name;
+    (void)value;
+
+    mStopProcessing = true;
+
+    std::cout << "Aos IA manager version:   " << AOS_CORE_SM_VERSION << std::endl;
```
Comment (mlohvynenko) [inline src/app/app.cpp:156]:
```suggestion
    std::cout << "Aos service manager version:   " << AOS_CORE_SM_VERSION << std::endl;
```

File: src/app/app.hpp [hunk 5]
```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
```
Comment (mlohvynenko) [inline src/app/app.hpp:2]:
```suggestion
```

File: src/app/aoscore.hpp [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+     * Starts Aos core.
+     */
+    void Start();
+
+    /**
+     * Stops Aos core.
+     */
+    void Stop();
+
+    aos::crypto::CertLoader                                              mCertLoader;
```
Comment (mykola-kobets-epam) [inline src/app/aoscore.hpp:57]: Init/Start/Stop should be public only, right?

**Review by mykola-kobets-epam — APPROVED** [review]

File: src/smclient/smclient.cpp [hunk 7]
```diff
@@ -66,22 +63,30 @@ Error SMClient::Start()
     if (mSecureConnection) {
         iam::certhandler::CertInfo certInfo;
 
-        if (auto err = mProvisionManager->GetCert(String(mConfig.mCertStorage.c_str()), {}, {}, certInfo);
-            !err.IsNone()) {
+        if (auto err = mTLSCredentials->GetCert(mConfig.mCertStorage.c_str(), {}, {}, certInfo); !err.IsNone()) {
```
[the remark this answers is not in the document]
Comment (al1img) [reply src/smclient/smclient.cpp]: fixed

File: src/smclient/smclient.cpp [hunk 8]
```diff
@@ -66,22 +63,30 @@ Error SMClient::Start()
...
             return AOS_ERROR_WRAP(Error(err, "can't get certificate"));
         }
 
-        if (auto err = mProvisionManager->SubscribeCertChanged(String(mConfig.mCertStorage.c_str()), *this);
-            !err.IsNone()) {
+        if (auto err = mTLSCredentials->SubscribeCertChanged(mConfig.mCertStorage.c_str(), *this); !err.IsNone()) {
```
[the remark this answers is not in the document]
Comment (al1img) [reply src/smclient/smclient.cpp]: fixed

File: src/smclient/smclient.cpp [hunk 9]
```diff
@@ -128,9 +133,17 @@ void SMClient::OnCertChanged(const iam::certhandler::CertInfo& info)
 
     LOG_INF() << "Certificate changed";
 
+    (void)info;
+
+    auto [creds, err] = mTLSCredentials->GetMTLSClientCredentials(mConfig.mCertStorage.c_str());
```
[the remark this answers is not in the document]
Comment (al1img) [reply src/smclient/smclient.cpp:132]: Please discuss with @mlohvynenko about separate thread. Retry mechanism should be done inside `GetMTLSClientCredentials` method IMHO.

File: src/app/CMakeLists.txt (hunk 1 above)
Comment (al1img) [reply src/app/CMakeLists.txt]: fixed

File: src/app/aoscore.cpp (hunk 2 above)
Comment (al1img) [reply src/app/aoscore.cpp]: fixed

File: src/app/aoscore.hpp (hunk 3 above)
Comment (al1img) [reply src/app/aoscore.hpp]: fixed

File: src/app/app.cpp (hunk 4 above)
Comment (al1img) [reply src/app/app.cpp]: fixed

File: src/app/app.hpp (hunk 5 above)
Comment (al1img) [reply src/app/app.hpp]: fixed

**Review by mlohvynenko — APPROVED** [review]

**Review by MykolaSolyanko — APPROVED** [review]

**Pull request merged**